src/listbox/Listbox.ts

```diff
@@ -0,0 +1,270 @@
+		const node = this.getNode(key);
+		if (node) {
+			node.scrollTop = amount;
+		}
+	}
+}
+
+/* Listbox Option sub-widget */
+export interface ListboxOptionProperties extends ThemeableProperties {
+	active?: boolean;
+	classes?: (string | null)[];
+	disabled?: boolean;
+	id: string;
+	index: number;
+	label: DNode;
+	option: any;
+	selected?: boolean;
+	onClick?(option: any, index: number, key?: string | number): void;
+}
+
+const ListboxOptionBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export class ListboxOption extends ListboxOptionBase<ListboxOptionProperties> {
+	private _onClick(event: MouseEvent) {
+		const { index, key, option, onClick } = this.properties;
+		onClick && onClick(option, index, key);
+	}
+
+	protected render() {
+		const {
+			classes = [],
+			disabled = false,
+			id,
+			label,
+			selected = false
+		} = this.properties;
+
+		return v('div', {
+			'aria-disabled': disabled ? 'true' : null,
+			'aria-selected': disabled ? null : String(selected),
+			classes: this.classes(...classes),
+			id,
+			role: 'option',
+			onclick: this._onClick
+		}, [ label ]);
+	}
+}
+
+/**
+ * @type ListboxProperties
+ *
+ * Properties that can be set on a Listbox component
+ *
+ * @property activeIndex          Index of the currently active listbox option
+ * @property describedBy          ID of an element that provides more descriptive text
+ * @property getOptionLabel       Function to return string label based on option data
+ * @property getOptionDisabled    Function that accepts option data and returns a boolean for disabled/not disabled
+ * @property getOptionId          Function that accepts option data and returns a string ID
+ * @property getOptionSelected    Function that accepts option data and returns a boolean for selected/unselected
+ * @property id                   Optional custom id for the listbox
+ * @property multiselect          Adds currect semantics for a multiselect listbox
+ * @property optionData           Array of data for listbox options
+ * @property tabIndex             Listbox is in the focus order by default, but setting tabIndex: -1 will remove it
+ * @property visualFocus          When controlling Listbox through an outside widget, e.g. in ComboBox, visualFocus mimics visual focus styling when true
+ * @property onActiveIndexChange  Called with the index of the new requested active descendant
+ * @property onOptionSelect       Called with the option data of the new requested selected item
+ */
+
+export interface ListboxProperties extends ThemeableProperties {
+	activeIndex?: number;
+	describedBy?: string;
+	getOptionDisabled?(option: any, index: number): boolean;
+	getOptionId?(option: any, index: number): string;
+	getOptionLabel?(option: any, index: number): DNode;
+	getOptionSelected?(option: any, index: number): boolean;
+	id?: string;
+	multiselect?: boolean;
+	optionData?: any[];
+	ScrollMeta?: WidgetMetaConstructor<DefaultScroll>;
```

Comment:
I just want to make sure it's easy to customize, e.g. to animate the scroll or something. Do you think there's anything wrong with passing it in as a property? I suppose just having `protected animateScroll` could be enough, since someone could override it.

Reply:
To me it feels somewhat overkill to provide both an extension point and an injection point. Should we not simply choose one pattern and stick to it?

Reply:
Works for me. I'm fine removing the prop and leaving `protected animateScroll` as the point of extension 👍